test/Microsoft.AspNetCore.Hosting.FunctionalTests/ShutdownTests.cs

```diff
@@ -16,6 +17,9 @@ namespace Microsoft.AspNetCore.Hosting.FunctionalTests
 {
     public class ShutdownTests : LoggedTest
     {
+        private static readonly Regex NowListeningRegex = new Regex(@"^\s*Now listening on: (?<url>.*)$");
```

Comment:
We could either make this public in SelfHostDeployer (which is an odd thing to expose) or make this internal visible to this test.